server/catalog/package_data_from_catalog.go

```diff
@@ -0,0 +1,95 @@
+package catalog
+
+import (
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis-package-indexer/server/types"
+	"github.com/kurtosis-tech/stacktrace"
+	"net/url"
+	"path"
+	"strings"
+)
+
+const (
+	githubDomain              = "github.com"
+	httpsSchema               = "https"
+	httpSchemaDomainSeparator = "://"
+	httpPathSeparator         = "/"
+	rootPathSeparator         = "/"
+)
+
+type packageDataFromCatalog struct {
+	// repositoryOwner is the repositoryOwner of the GitHub repository. It can be a GitHub organization or an individual user
+	repositoryOwner string
+
+	// repositoryName is the name of the GitHub repository
+	repositoryName string
+
+	// repositoryPackageRootPath is the path relative to the GitHub repository root where the kurtosis.yml file can be found
+	repositoryPackageRootPath string
+}
+
+func (mainMetadata *packageDataFromCatalog) GetRepositoryOwner() string {
+	return mainMetadata.repositoryOwner
+}
+
+func (mainMetadata *packageDataFromCatalog) GetRepositoryName() string {
+	return mainMetadata.repositoryName
+}
+
+func (mainMetadata *packageDataFromCatalog) GetRepositoryPackageRootPath() string {
+	return mainMetadata.repositoryPackageRootPath
+}
+
+func createPackageDataFromCatalogFromPackageName(packageName types.PackageName) (*packageDataFromCatalog, error) {
+	// we expect something like github.com/repository-repositoryOwner/repository-name
+	// we don't want schemas
+	parsedURL, err := url.Parse(string(packageName))
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "an error occurred parsing the URL of package '%s'", packageName)
+	}
+	if parsedURL.Scheme != "" {
+		return nil, stacktrace.NewError("an error occurred parsing the URL of module '%v'. Expected schema to be empty got '%v'", packageName, parsedURL.Scheme)
+	}
+
+	// we prefix schema and make sure that the URL still parses
+	packageURLPrefixedWithHttps := fmt.Sprintf("%s%s%s", httpsSchema, httpSchemaDomainSeparator, packageName)
+	parsedURL, err = url.Parse(packageURLPrefixedWithHttps)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "an error occurred parsing the URL with scheme '%s' for package '%s'", packageURLPrefixedWithHttps, packageName)
+	}
+	if strings.ToLower(parsedURL.Host) != githubDomain {
+		return nil, stacktrace.NewError("an error occurred parsing the URL of package. We only support packages on Github for now but got '%v'", packageName)
+	}
+
+	splitPath := cleanPathAndSplit(parsedURL.Path)
+
+	repositoryOwner := splitPath[0]
+	repositoryName := splitPath[1]
```

Comment:
is it possible this could yield a NIL panic?

Reply:
yes, if for some reason one row in the catalog is wrong that could happen, good catch, let me add a checker